### DISCUSSION TITLE
Custom Filter Dialog

### DISCUSSION THREAD

**Discussion by dg943** [serenity-is/Serenity#6757] [Help]
Hey, Everyone

I wanted to have a custom filter dialog which will get opened when I click on the filter icon at the bottom right of the grid page, Is it feasible currently in serenity?

**Comment by gfo2007** [comment]
```
@Decorators.filterable()
```

in the xxxgrid.ts

**Comment by dg943** [comment]
Thanks @gfo2007 for your quick response.
But I think by adding this line we will get the filter option at the bottom of the grid. and after clicking on this it will open FilterDialog which is something inbuilt in serenity, But I want to open my own dialog which will extend FilterDialog.

Need for my own custom dialog => wanted to have some extra buttons and functionalities on it.

**Reply by dg943** [reply]
@volkanceylan please provide your input.

**Reply by VictorTomaili** [reply]
There is no easy way to do this.
The first option is using scriptinit to override the required function to open your extended dialog instead of the default one. 
The other option is, analyzing existing filter dialog and implement own filter dialog with your new decorator.

**Reply by dg943** [reply]
Thanks @VictorTomaili, that is exactly what I wanted to know.
